### PR TITLE
NN-3799: New endpoint to provide paginated search for all completed reported adjudications

### PR DESCRIPTION
**Pull request by sp-watson**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/DraftAdjudicationService.kt [hunk 1]
```diff
@@ -115,6 +115,8 @@ class DraftAdjudicationService(
     submittedAdjudicationHistoryRepository.save(
       SubmittedAdjudicationHistory(
         adjudicationNumber = reportedAdjudication.adjudicationNumber,
+        agencyId = reportedAdjudication.agencyId,
```
Comment (sp-watson) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/DraftAdjudicationService.kt:118]: We need to store these so we can search on them

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/IntTestData.kt [hunk 2]
```diff
@@ -1,8 +1,20 @@
 package uk.gov.justice.digital.hmpps.hmppsmanageadjudicationsapi.integration
 
+import org.springframework.http.HttpHeaders
```
Comment (sp-watson) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/IntTestData.kt:3]: This class, and AdjIntTestData, are a way of providing test data that is consistent between the API DB and Prison API.

It probably needs some refactoring, but IMO will make it far quicker to write int tests that involve the Prison API and the DB

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/ReportedAdjudicationIntTest.kt [hunk 3]
```diff
@@ -101,6 +101,46 @@ class ReportedAdjudicationIntTest : IntegrationTestBase() {
       .jsonPath("$.content[1].incidentDetails.locationId").isEqualTo(721850)
   }
 
+  @Test
+  fun `return a page of reported adjudications completed in the current agency`() {
+    val intTestData = IntTestData(webTestClient, jwtAuthHelper, bankHolidayApiMockServer, prisonApiMockServer)
```
Comment (sp-watson) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/ReportedAdjudicationIntTest.kt:106]: We could probably abstract this into the TestBase class, but that's a future change

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/wiremock/PrisonApiMockServer.kt [hunk 4]
```diff
@@ -125,6 +141,43 @@ class PrisonApiMockServer : WireMockServer(8979) {
     )
   }
 
+  fun stubGetValidAdjudicationsById(testDataSet1: AdjudicationIntTestDataSet, testDataSet2: AdjudicationIntTestDataSet) {
```
Comment (sp-watson) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/wiremock/PrisonApiMockServer.kt:144]: I could have made this more generic and reusable by passing a list of data sets, but IMO that is going too much in to the realms of writing another app.

Best to minimise the magic in tests.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/wiremock/PrisonApiMockServer.kt [hunk 5]
```diff
@@ -17,7 +21,19 @@ import uk.gov.justice.digital.hmpps.hmppsmanageadjudicationsapi.integration.IntT
 import uk.gov.justice.digital.hmpps.hmppsmanageadjudicationsapi.integration.IntTestData.Companion.UPDATED_LOCATION_ID
 import uk.gov.justice.digital.hmpps.hmppsmanageadjudicationsapi.integration.IntTestData.Companion.UPDATED_STATEMENT
 
-class PrisonApiMockServer : WireMockServer(8979) {
+class PrisonApiMockServer : WireMockServer {
+  constructor() : super(8979) {
```
Comment (sp-watson) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/wiremock/PrisonApiMockServer.kt:25]: This logging was really useful in detecting errors in mocking data, so I have kept it in.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/DraftAdjudicationService.kt (hunk 1 above)
Comment (richardpopple) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/DraftAdjudicationService.kt:118]: Cool - I guess as time progresses we might need to add more data

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/IntTestData.kt (hunk 2 above)
Comment (richardpopple) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/IntTestData.kt:3]: Coolio

**Review by richardpopple — APPROVED** [review]

**Pull request merged**